File: P389 Find the Difference/Solution.py
class Solution(object):
    def findTheDifference(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: str
        """
        cs = collections.Counter(s)
        for letter in t:
            cs[letter] -= 1
            if cs[letter] <0: return letter
        return ''

File: P389 Find the Difference/Solution.cpp
class Solution {
public:
    char findTheDifference(string s, string t) {
        vector<int> v(26, 0);  
        for (int i = 0; i < s.length(); ++i)  
            ++v[s[i]-97];  
  
        for (int i = 0; i < t.length(); ++i)  
        {  
            --v[t[i]-97];  
            if (v[t[i]-97] < 0)  
                return t[i];  
        }//for  
        return ' ';    
    }
};